python/ray/parameter.py

```diff
@@ -0,0 +1,192 @@
+                 num_gpus=None,
+                 num_local_schedulers=None,
+                 resources=None,
+                 object_store_memory=None,
+                 redis_max_memory=None,
+                 redis_port=None,
+                 redis_shard_ports=None,
+                 object_manager_ports=None,
+                 node_manager_ports=None,
+                 collect_profiling_data=True,
+                 node_ip_address=None,
+                 object_id_seed=None,
+                 num_workers=None,
+                 local_mode=False,
+                 driver_mode=None,
+                 redirect_worker_output=False,
+                 redirect_output=True,
+                 num_redis_shards=None,
+                 redis_max_clients=None,
+                 redis_password=None,
+                 plasma_directory=None,
+                 worker_path=None,
+                 huge_pages=False,
+                 include_webui=True,
+                 logging_level=logging.INFO,
+                 logging_format=ray_constants.LOGGER_FORMAT,
+                 plasma_store_socket_name=None,
+                 raylet_socket_name=None,
+                 temp_dir=None,
+                 include_log_monitor=None,
+                 autoscaling_config=None,
+                 _internal_config=None):
+        self.address_info = address_info
+        self.start_ray_local = start_ray_local
+        self.object_id_seed = object_id_seed
+        self.redis_address = redis_address
+        self.num_cpus = num_cpus
+        self.num_gpus = num_gpus
+        self.num_local_schedulers = num_local_schedulers
+        self.resources = resources
+        self.object_store_memory = object_store_memory
+        self.redis_max_memory = redis_max_memory
+        self.redis_port = redis_port
+        self.redis_shard_ports = redis_shard_ports
+        self.object_manager_ports = object_manager_ports
+        self.node_manager_ports = node_manager_ports
+        self.collect_profiling_data = collect_profiling_data
+        self.node_ip_address = node_ip_address
+        self.num_workers = num_workers
+        self.local_mode = local_mode
+        self.driver_mode = driver_mode
+        self.redirect_worker_output = redirect_worker_output
+        self.redirect_output = redirect_output
+        self.num_redis_shards = num_redis_shards
+        self.redis_max_clients = redis_max_clients
+        self.redis_password = redis_password
+        self.plasma_directory = plasma_directory
+        self.worker_path = worker_path
+        self.huge_pages = huge_pages
+        self.include_webui = include_webui
+        self.plasma_store_socket_name = plasma_store_socket_name
+        self.raylet_socket_name = raylet_socket_name
+        self.temp_dir = temp_dir
+        self.include_log_monitor = include_log_monitor
+        self.autoscaling_config = autoscaling_config
+        self._internal_config = _internal_config
+
+    def apply_settings(self, **kwargs):
```

Comment:
I have a slight preference to rename this function to `update`, because it's analog to `dict.update`. Accordingly, the other method could be `update_if_absent`.